### ISSUE TITLE
`sits_mae()` fails on large models due to internal `torch_save()` serialization of long raw vector

### ISSUE THREAD

**Issue by rolfsimoes** [e-sensing/sits#1460]
**Describe the bug**
Training a model with `sits_mae()` on large datasets fails with the error:

```r
Error in torch_tensor_cpp(data, dtype, device, requires_grad, pin_memory) : 
  long vectors not supported yet: memory.c:3948
```

The error occurs at the end of training, when `sits_mae()` returns the trained model. The training process itself (forward pass, loss computation, dataloader) completes normally.

**To Reproduce**

```r
library(sits)

samples <- readRDS("~/big_samples.rds")

mae_model <- sits_pre_train(
  samples = samples,
  encoder_method = sits_mae(
      encoder_model = sits_lighttae(),
      mask_ratio = 0.5,
      epochs = 1,
      verbose = TRUE
  )
)
```

Dataset characteristics:

* Number of samples: 72,500
* Time steps per sample: 46
* Number of bands: 10

The error occurs consistently with larger datasets.

**Additional context**

The failure happens during the construction of the object returned by `sits_mae()`.

By debugging the code path, the error was traced to the serialization step used internally by `torch::torch_save()`.

In `torch`, `torch_save.nn_module()` calls `torch_save_to_file()`, which includes:

```r
state_dict[[r_obj]] <- torch_tensor(serialize(object, connection = NULL))
```

This serializes the full R object into an in-memory raw vector and then converts it to a torch tensor.

For large models, the serialized object becomes large enough to trigger an internal code path that does not support long vectors, resulting in:

```r
long vectors not supported yet: memory.c:3948
```

This explains why training completes successfully, but the failure occurs at the final serialization step in `sits`.

Possible approach:

Avoid serializing the full model into a single large object and store only the `state_dict()` instead of the full model.
